students/Kate_Koehn/Lesson06/good_perf.py

```diff
@@ -0,0 +1,85 @@
+"""
+This module is the cleaned up version of poor_perf.py. The file is only read and iterated over a single time,
+and all operations that need to be performed on each row have been consolidated into a single for loop.
+"""
+from timeit import timeit
+import datetime
+import csv
+
+
+def analyze(filename):
+    """
+    Program to read in a csv, loop over the rows in the file, convert the date from a string to a datetime object,
+    save count of years 2013-2018 in a dict, and count how many times "ao" appears in the file.
+    """
+
+    # timestamp for when we start executing the function
+    start = datetime.datetime.now()
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+
+        # skip header
+        next(reader, None)
+        # we only care about dates after 01/01/2012, this line converts that date to a datetime object to be able
+        # to compare to other dates in the file
+        date =  datetime.datetime.strptime('01/01/2012', '%m/%d/%Y')
+        # save strptime as a variable that we can call on dates in the file later to convert them to datetime objects
+        my_date = datetime.datetime.strptime
+
+        # counter for 'ao's in the file
+        found = 0
+        # counter dict for each year we care about counting in the file
+        year_count = {
+            "2013": 0,
+            "2014": 0,
+            "2015": 0,
+            "2016": 0,
+            "2017": 0,
+            "2018": 0
+        }
```

Comment:
I don't know if it is more efficient, but fun place for a dictionary comprehension:
```
year_count = {str(year): 0 for year in range(2013, 2019)}
```